### ISSUE TITLE
Roadmap: propose what you'd like to see

### ISSUE THREAD

**Issue by Zughy** [LibreSprite/LibreSprite#160]
As the development is back on track (just compare commits of this month with the other ones), a roadmap can help us avoid derailing and getting lost. As discussed on Discord/Matrix, each maintainer/contributor/user can list 3 points about what they think should be prioritised  for the 1.0 release (currently LibreSprite says "1.0-dev"), so to have a wider perspective of both users needs and development musts.
This is **not** a contest, meaning that the most voted feature won't necessarily be implemented (e.g. development issues, troll votes). Because of that, maintainers reserve the final decision.

I'll start:
- **[Back-end] get rid of Allegro 4 and use SDL2 instead**: right now there are bugs we simply can't solve, because they're part of the current back-end (Allegro 4). Aseprite stuck with Skia, but everybody seems to somehow dislike it (including me). According to an SDL2 expert (https://github.com/LibreSprite/LibreSprite/issues/2#issuecomment-906817056), this last one is not just viable, but even better than Skia. As for now, @felipemanga is doing some testing in his own fork https://github.com/felipemanga/LibreSprite/tree/sdl2-backend. If this works, it'll mean having a way easier life both for maintainers and contributors (with users benefiting consequently).
- **Mac support**: right now LibreSprite only works on Linux and Windows. The more (famous) OS we can cover, the more contributors we can have. This needs the previous point first, as it doesn't work with Allegro 4.
- **More default palettes**: as the first two points are technical, I've felt like leaving the last one more artistic-driven. LibreSprite doesn't currently have so many artistic palettes, and something like #130 can make a lot of artists (amateurs and not) very happy

**Comment by Torbuntu** [comment]
A switch to SDL2 would be great! I've always struggled (never even succeeded) with building Aseprite because of Skia.

**Comment by felipemanga** [comment]
- As someone who's had headaches due to Skia, I'd like to reiterate the need for an SDL2 backend.
- The code could use a good general clean-up. It seems that the current structure will not grow sustainably as more features are added.
- Layer groups and masks.

**Comment by jjhaggar** [comment]
I'll list the ones I think that woud make LibreSprite more powerful (I have not much idea about the technical ones, so I'll focus only on features I can understand):
- Mac support (and Raspberry Pi if possible).
- PSD importing and exporting (I think that these features would probably require layer groups and masks).
- Script compatibility with [Aseprite's Lua scripts](https://www.aseprite.org/docs/scripting/). I'm not sure if this can be done, but there are a lot of scripts out there that could be extremely useful for LibreSprite users (for example, there is one that export ASE files to PSD format).

**Comment by Torbuntu** [comment]
>  (and Raspberry Pi if possible).

If you use an aarch64 distro on a pi3 or higher, you can use the flatpak like I do from here https://flathub.org/apps/details/com.github.libresprite.LibreSprite

I even have it on my pinephone and pinetab.

**Comment by Charl420** [comment]
Make multiple official themes in the files that you install to choose what theme to use, like: light, dark, pink, blue, rainbow and red.

**Comment by maujin111** [comment]
> > (and Raspberry Pi if possible).
> 
> If you use an aarch64 distro on a pi3 or higher, you can use the flatpak like I do from here https://flathub.org/apps/details/com.github.libresprite.LibreSprite
> 
> I even have it on my pinephone and pinetab.

Thanks a lot!!!

**Comment by Zughy** [comment]
>     * The code could use a good general clean-up. It seems that the current structure will not grow sustainably as more features are added.

I agree that it needs clean-up, but I think it's something you do whilst messing around with it. Every time I try to implement or fix something, I discover questionable parts of code, cleaning them up whenever I can

>     * Script compatibility with [Aseprite's Lua scripts](https://www.aseprite.org/docs/scripting/).

Quoting @felipemanga on Matrix/Discord: "supporting both languages [JS and Lua] is a pain, but doable. Supporting a compatible API... that's going to be a much bigger pain". Because it'd mean following everything Aseprite does, after catching up 5 years of development, and keep adapting consequently

**Comment by maujin111** [comment]
I think LibreSprite should also be able to support LUA scripts or any other easy programming language for the end user

**Comment by jjhaggar** [comment]
> Quoting @felipemanga on Matrix/Discord: "supporting both languages [JS and Lua] is a pain, but doable. Supporting a compatible API... that's going to be a much bigger pain". Because it'd mean following everything Aseprite does, after catching up 5 years of development, and keep adapting consequently

Yeah, that makes sense, it's a shame but it looks like too much work. Well, we could maybe try to support both Javascript and Lua scripts. Even if we don't adapt to the exact same API Aseprite uses for Lua scripts, I think chances are very high that they will naturally have some similarities, hence adapting the existent scripts should be more or less easy :)

From what I remember, most of the Aseprite scripts I've seen have a GPL or MIT license, so I think that in the future, we maybe could curate them and translate & add the best ones to LibreSprite by default (they don't take much disk space XD), or maybe making an easy way to install them (like clicking a button, or something similar).

I made a list of Aseprite scripts here: https://community.aseprite.org/t/aseprite-scripts-collection/3599 (I was planning to make a repository collecting and organizing them, but I never had enough time)
